packages/data-table-server/src/dataTableService/DataTableService.ts

```diff
@@ -3,28 +3,28 @@
  * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
  */
 
-import { TupaiaApiClient } from '@tupaia/api-client';
 import { yup } from '@tupaia/utils';
 
 export abstract class DataTableService<
+  Context extends Record<string, unknown> = Record<string, unknown>,
   ParamsSchema extends yup.AnySchema = yup.AnySchema,
   ConfigSchema extends yup.AnySchema = yup.AnySchema,
   RecordSchema = unknown
 > {
+  protected readonly ctx: Context;
```

Comment:
This always surprises me, and seems like a bug, but eslint complains if you call a field of a class `context` with the following [error](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/static-property-placement.md).

I guess I don't mind too much, but it is always a little surprising. Perhaps we look into this? Perhaps we choose a different term other than `context` to refer to arbitrary state?